libcontainer/rootfs_linux.go

```diff
@@ -1068,19 +1068,58 @@ func pivotRoot(rootfs string) error {
 	}
 	defer unix.Close(oldroot) //nolint: errcheck
 
-	newroot, err := unix.Open(rootfs, unix.O_DIRECTORY|unix.O_RDONLY, 0)
-	if err != nil {
-		return &os.PathError{Op: "open", Path: rootfs, Err: err}
-	}
-	defer unix.Close(newroot) //nolint: errcheck
-
 	// Change to the new root so that the pivot_root actually acts on it.
-	if err := unix.Fchdir(newroot); err != nil {
-		return &os.PathError{Op: "fchdir", Path: "fd " + strconv.Itoa(newroot), Err: err}
+	if err := os.Chdir(rootfs); err != nil {
+		return err
```

Comment:
I wondered why we did `fd = open(dir); fchdir(fd);` instead of just `chdir(dir)`, found it was introduced by commit f8e6b5af. My best guess it just followed the logic of some code (lxc?) which did it the same way.

Can't see if one way is better than another, but chdir of course is simpler and is one syscall instead of 3, so yay to simplicity!

Reply:
I think an older version of that patch did actually use the path more than once, so opening it made sense, but then we ended up with the version that doesn't so it was unnecessary.